ui/lib/pki/addon/routes/roles/index.js

```diff
@@ -0,0 +1,24 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class RolesIndexRoute extends Route {
+  @service store;
+  @service secretMountPath;
+  @service pathHelp;
+
+  async model() {
+    // the pathHelp service is needed for adding openAPI to the model
+    await this.pathHelp.getNewModel('pki/pki-role-engine', 'pki');
```

Comment:
You normally don't see this method being called in other secret engines because it's done automatically when visiting the backend.list-root route, however we're not visiting that route because of the mounting of where this engine is. Therefor we have to move some functionality the list-route gives us to this model.